its/it/azdo.ts

```diff
@@ -9,6 +22,46 @@ export function getAzdoApi(): vm.WebApi {
   return new vm.WebApi(AZDO_BASE_URL + AZDO_ORGANIZATION, credentialHandler);
 }
 
+export async function getTemplatePipeline(buildApi: IBuildApi): Promise<BuildDefinition> {
+  // Find an existing pipeline to use as a template
+  const definitions = await buildApi.getDefinitions(AZDO_PROJECT);
+  const existingDefinition = definitions.find((d) => d.name?.startsWith(AZDO_PIPELINE_NAME_PREFIX));
+  if (!existingDefinition) {
+    throw new Error("No existing pipeline found to use as a template");
```

Comment:
this will fail the task right? which would indicate the pipeline hasn't been generated? 

It's not clear to me as the step above looks like you search all definitions (pipelines) from the project that start with `pipeline-` which IMO would be most of them no? I must be missing something

Reply:
> to copy its definition for the new pipelines

the definitions today are created during the generate template task, so we can trust copying any created template to copy definitions? I added a comment where we spread the template, I'm worried this might result in unforeseen consequences 🤔